TABLE_NOTES.md

```diff
@@ -0,0 +1,106 @@
+# Thoughts on table control support
+
+(BenC - 2017-09-25)
+
+## win32 comctl listview TODOs
+
+- maintain proper list of column parts, even if dummy.
+   (or cut down the public API to text columns only for now
+   until a better windows table control is available - see below
+   for more thoughts on this)
+- support editing of cells
+- expandable parts (allocate leftover space)
+- how to determine column width? random sample of data?
+
+
+## Is the parts system the right abstraction?
+
+Some toolkits support implementing cell views as custom
+widget layouts eg [item delgates in QML](http://doc.qt.io/qt-5/qml-qtquick-controls-tableview.html#itemDelegate-prop).
+So an item can be just about any widget layout you want.
+
+Is that kind of thing a future aspiration for libui?
+If so, I'd suggest cutting the parts API back to just
+uiTableAppendTextColumn() for now.
+
+The parts API gives _some_ of the flexiblity, but would likely
+get in the way if libui went the whole hog on custom item view
+layouts.
+
+
+## API change: support bulk invalidation of model
+
+Currently, the only way to notify the control that new data has
+been added is via the uiTableModelRow[Inserted|Changed|Deleted]
+functions. This inevitably causes redraws for each item, which
+can get _really_ slow for large numbers of items.
+
+The use cases are:
+ - adding large numbers of rows to existing models
+ - sorting (eg clicking on various column headings to change
+   the sort field and order.
+   The user-supplied model needs to handle the sorting, the
+   table control can't help.
+
+I think a single uiTableModelAllChanged() function would be
+enough. The underlying control needs to know that it should
+to redraw everything visible (and the number of items might
+have chnaged).
```

Comment:
As noted, the problem is GTK+ does not seem to provide a way to do this. There might be workarounds that are inefficient, and I'm not sure if removing and re-adding the model will reset other settings like column widths and scroll positions. Will need to ask the GTK+ people about it.

Might want to note this here. Not required; not part of review.